notebooks/working/investigating_deconfounder.py

```diff
@@ -338,28 +309,39 @@
     ax.axvline(x=obs_ll_per_zi[subject_no], linestyle='--')
 
 plt.show()
-# -
 
-# ## Visualizing the substitute confounder against the true confounder
+# +
+overall_ll_est = rep_ll_per_zi.mean(axis=1)
 
-df.plot(kind='scatter', x = 'confounder_PCA_SKLEARN', y = 'confounder')
+overall_ll_obs = obs_ll_per_zi.mean()
 
-df.plot(kind='scatter', x = 'confounder_PPCA', y = 'confounder')
+sns.kdeplot(overall_ll_est).set_title("Predictive check: Population")
+plt.axvline(x=overall_ll_obs, linestyle='--')
```

Comment:
Can you add some text under this plot to let readers know that this is yet another posterior predictive check of the factor model's log-likelihood, and the predictive check fails terribly?

Just point out that a model should not be strongly refuted by any posterior predictive checks, and if it is, it should cause one to further investigate one's model performance (even if another posterior predictive check has passed).